src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -512,6 +527,10 @@ private async ValueTask FlushSendBufferAsync(bool endStream, CancellationToken c
         {
             await _stream.WriteAsync(_sendBuffer.ActiveMemory, endStream, cancellationToken).ConfigureAwait(false);
             _sendBuffer.Discard(_sendBuffer.ActiveLength);
+            if (endStream)
+            {
+                await _stream.WritesClosed.ConfigureAwait(false);
+            }
 
             await _stream.FlushAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
It feels odd that we're flushing after we've ensured that writes are complete. If the flush is a no-op can we delete it?

Reply:
I'll switch the order.